accounts/abi/bind/backends/simulated.go

```diff
@@ -196,6 +204,123 @@ func (b *SimulatedBackend) TransactionByHash(ctx context.Context, txHash common.
 	return nil, false, ethereum.NotFound
 }
 
+// TransactionByHash returns the transaction with the given hash.
+func (b *SimulatedBackend) TransactionByHashWithBlockNum(ctx context.Context, txHash common.Hash) (tx *types.Transaction, blockHexString string, err error) {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	tx = b.pendingBlock.Transaction(txHash)
+	if tx != nil {
+		return tx, hexutil.EncodeUint64(0), nil
+	}
+
+	tx, _, blockNumber, _ := rawdb.ReadTransaction(b.database, txHash)
+	if tx != nil {
+		return tx, hexutil.EncodeUint64(blockNumber), nil
```

Comment:
why not return uint64 here?

Reply:
hopefully we can just remove this function